core/vm/evmc.go

```diff
@@ -0,0 +1,354 @@
+		Coinbase:   host.env.Coinbase,
+		Number:     host.env.BlockNumber.Int64(),
+		Timestamp:  host.env.Time.Int64(),
+		GasLimit:   int64(host.env.GasLimit),
+		Difficulty: common.BigToHash(host.env.Difficulty)}
+}
+
+func (host *hostContext) GetBlockHash(number int64) common.Hash {
+	b := host.env.BlockNumber.Int64()
+	if number >= (b-256) && number < b {
+		return host.env.GetHash(uint64(number))
+	}
+	return common.Hash{}
+}
+
+func (host *hostContext) EmitLog(addr common.Address, topics []common.Hash, data []byte) {
+	host.env.StateDB.AddLog(&types.Log{
+		Address:     addr,
+		Topics:      topics,
+		Data:        data,
+		BlockNumber: host.env.BlockNumber.Uint64(),
+	})
+}
+
+func (host *hostContext) Call(kind evmc.CallKind,
+	destination common.Address, sender common.Address, value *big.Int, input []byte, gas int64, depth int,
+	static bool, salt *big.Int) (output []byte, gasLeft int64, createAddr common.Address, err error) {
+
+	gasU := uint64(gas)
+	var gasLeftU uint64
+
+	switch kind {
+	case evmc.Call:
+		if static {
+			output, gasLeftU, err = host.env.StaticCall(host.contract, destination, input, gasU)
+		} else {
+			output, gasLeftU, err = host.env.Call(host.contract, destination, input, gasU, value)
+		}
+	case evmc.DelegateCall:
+		output, gasLeftU, err = host.env.DelegateCall(host.contract, destination, input, gasU)
+	case evmc.CallCode:
+		output, gasLeftU, err = host.env.CallCode(host.contract, destination, input, gasU, value)
+	case evmc.Create:
+		var createOutput []byte
+		createOutput, createAddr, gasLeftU, err = host.env.Create(host.contract, input, gasU, value)
+		isHomestead := host.env.ChainConfig().IsHomestead(host.env.BlockNumber)
+		if !isHomestead && err == ErrCodeStoreOutOfGas {
+			err = nil
+		}
+		if err == errExecutionReverted {
+			// Assign return buffer from REVERT.
+			// TODO: Bad API design: return data buffer and the code is returned in the same place. In worst case
+			//       the code is returned also when there is not enough funds to deploy the code.
+			output = createOutput
+		}
+	case evmc.Create2:
+		var createOutput []byte
+		createOutput, createAddr, gasLeftU, err = host.env.Create2(host.contract, input, gasU, value, salt)
+		if err == errExecutionReverted {
+			// Assign return buffer from REVERT.
+			// TODO: Bad API design: return data buffer and the code is returned in the same place. In worst case
+			//       the code is returned also when there is not enough funds to deploy the code.
+			output = createOutput
+		}
+	default:
+		panic(fmt.Errorf("EVMC: Unknown call kind %d", kind))
+	}
+
+	// Map errors.
+	if err == errExecutionReverted {
+		err = evmc.Revert
+	} else if err != nil {
+		err = evmc.Failure
+	}
+
+	gasLeft = int64(gasLeftU)
+	return output, gasLeft, createAddr, err
+}
+
+// getRevision translates ChainConfig's HF block information into EVMC revision.
+func getRevision(env *EVM) evmc.Revision {
+	n := env.BlockNumber
+	conf := env.ChainConfig()
```

Comment:
Is there Istanbul here yet?

Reply:
Now yes

Reply:
Meanwhile, EVMC v6 doesn't offer proper Istanbul support, so better to leave it out :)